src/nth.rs

```diff
@@ -37,12 +38,17 @@ pub fn parse_nth(input: &mut Parser) -> Result<(i32, i32), ()> {
                 }
             }
         }
-        Token::Delim('+') => match try!(input.next_including_whitespace()) {
-            Token::Ident(value) => {
-                match_ignore_ascii_case! { &value,
-                    "n" => parse_b(input, 1),
-                    "n-" => parse_signless_b(input, 1, -1),
-                    _ => Ok((1, try!(parse_n_dash_digits(&*value))))
+        Token::Delim('+')| Token::Delim('-') => match try!(input.next_including_whitespace()) {
```

Comment:
nit: spacing.

But I'd write it as:

```rust
Token::Delim(delim) => {
    let val = match delim {
        '+' => 1,
        '-' => -1,
        _ => return Err(()),
    };
    match try!(input.next_including_whitespace()) {
        // ...
    }
}
```

That way you avoid the extra variable at the top too.